src/App.js

```diff
@@ -15,27 +20,25 @@ function App() {
 		setCart([...cart, item]);
 	};
 
+	const removeItem = item => {
+		setCart(cart.filter(product => product.id !== item.id));
+	};
+
 	return (
-		<div className="App">
-			<Navigation cart={cart} />
-
-			{/* Routes */}
-			<Route
-				exact
-				path="/"
-				render={() => (
-					<Products
-						products={products}
-						addItem={addItem}
-					/>
-				)}
-			/>
-
-			<Route
-				path="/cart"
-				render={() => <ShoppingCart cart={cart} />}
-			/>
-		</div>
+		<ProductContext.Provider value={{ products, addItem }}>
+			<CartContext.Provider value={{ cart, removeItem }}>
+				<div className="App">
+						<Navigation cart={cart} />
```

Comment:
You don't need to pass the cart prop to `Navigation` since we are getting this data from the `useContext` hook inside of `Navigation.js`